### PR TITLE
Fix: Agent Reports counts when they have access to multiple accounts

### LINKED ISSUE #4660: bug: Agent Reports counts are broken when they have access to multiple accounts
Image

Agent overview reports seems to be displaying data across all the accounts at the moment. We want to scope it to the specific account which administrator is viewing at the moment. 

## Fix

https://github.com/chatwoot/chatwoot/blob/f64cf85ab24e933e9469d0eab45c748f4f859f3e/app/helpers/report_helper.rb#L20

should be called with 
```
(get_grouped_values scope.conversations.where(account_id: @account).count
```

- should ensure that all the other scopes in this helper is also properly account scoped.

### PR DESCRIPTION
**Pull request by aswindevps**
## Description

This change restricts the agent report API to fetch agent metrics from the current account. 

Fixes #4660

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

https://www.loom.com/share/080f1d64e13c4acaa8025c1bcb992ce9

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: spec/controllers/api/v2/accounts/report_controller_spec.rb [hunk 1]
```diff
@@ -223,6 +223,39 @@
...
+      let(:params) do
+        super().merge(
+          type: :agent,
+          since: 30.days.ago.to_i.to_s,
+          until: date_timestamp.to_s
+        )
+      end
+
+      it 'returns agent metrics from the current account' do
+        admin1 = create(:user, account: account1, role: :administrator)
```
Comment (sojan-official) [inline spec/controllers/api/v2/accounts/report_controller_spec.rb:240]: we haven't added admin1 as a member of account 2 here I think. 
we have to add
```
create(:account_user, user: admin1, account: account2)
```

**Review by tejaswinichile — COMMENTED** [review]
One small doubt, otherwise looks good to me.

File: spec/controllers/api/v2/accounts/report_controller_spec.rb [hunk 2]
```diff
@@ -223,6 +223,39 @@
         expect(response).to have_http_status(:success)
       end
     end
+
+    context 'when an agent has access to multiple accounts' do
+      let(:account1) { create(:account) }
+      let(:account2) { create(:account) }
+
+      let(:params) do
+        super().merge(
```
Comment (pranavrajs) [inline spec/controllers/api/v2/accounts/report_controller_spec.rb:232]: For my understanding, what does this do?

Comment (sojan-official) [reply spec/controllers/api/v2/accounts/report_controller_spec.rb:232]: there is a params object defined in the parent's block.. super will access that and then the additional params will be merged into it

File: app/helpers/report_helper.rb [hunk 3]
```diff
@@ -17,38 +17,38 @@ def scope
   end
 
   def conversations_count
-    (get_grouped_values scope.conversations).count
+    (get_grouped_values scope.conversations.where(account_id: account.id)).count
```
[the remark this answers is not in the document]
Comment (aswindevps) [reply app/helpers/report_helper.rb:20]: @sojan-official any thoughts here?

File: spec/controllers/api/v2/accounts/report_controller_spec.rb (hunk 1 above)
Comment (aswindevps) [reply spec/controllers/api/v2/accounts/report_controller_spec.rb:240]: Done

File: app/helpers/report_helper.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (sojan-official) [reply app/helpers/report_helper.rb:20]: the link @tejaswinichile associated with the line isn't redirecting me to a specific line.  😅 .  So I am not sure. 
Would love to keep the account scoping in a unified place. But I don't mind taking it as refactoring for the future.

**Pull request merged**